config/initializers/menus.rb

```diff
@@ -32,7 +32,15 @@
 Redmine::MenuManager.map :top_menu do |menu|
   # projects menu will be added by
   # Redmine::MenuManager::TopMenuHelper#render_projects_top_menu_node
-
+  
+  menu.push :projects,
+            { controller: '/projects', project_id: nil, action: 'index' },
+            context: :modules,
+            caption: I18n.t('label_projects_menu'),
+            if: Proc.new {
+              (User.current.logged? || !Setting.login_required?) &&
+                User.current.allowed_to?(:view_projects, nil, global: true)
```

Comment:
The permission `view_projects` does not exist. The reason you saw the menu entry is because you were logged in as an admin user, for which this method always returns true: https://github.com/opf/openproject/blob/dev/app/services/authorization/user_allowed_service.rb#L103

Thus, for any non-admin, this menu entry is never shown. So you can simply remove the second condition.